episodes/superspreading-estimate.Rmd

```diff
@@ -0,0 +1,836 @@
+
+:::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::: spoiler
+
+### Poisson, overdispersion, and Negative Binomial
+
+<!-- distribution stories -->
+
+For occurrences of associated discrete events we can use **Poisson** or negative binomial discrete distributions.
+
+In a Poisson distribution, mean is equal to variance. But when variance is higher than the mean, this is called **overdispersion**. In biological applications, overdispersion occurs and so a negative binomial may be worth considering as an alternative to Poisson distribution.
+
+**Negative binomial** distribution is specially useful for discrete data over an unbounded positive range whose sample variance exceeds the sample mean. In such terms, the observations are overdispersed with respect to a Poisson distribution, for which the mean is equal to the variance.
+
+In epidemiology, [negative binomial](https://en.wikipedia.org/wiki/Negative_binomial_distribution) have being used to model disease transmission for infectious diseases where the likely number of onward infections may vary considerably from individual to individual and from setting to setting, capturing all variation in infectious histories of individuals, including properties of the biological (i.e. degree of viral shedding) and environmental circumstances (e.g. type and location of contact).
+
+:::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::: challenge
+
+Calculate the distribution of secondary cases for Ebola using the `ebola_sim_clean` object from `{outbreaks}` package.
+
+Does the offspring distribution of Ebola is skewed or overdispersed?
+
+:::::::::::::::::: hint
+
+**Note:** This dataset has `r nrow(ebola_sim_clean$linelist)` cases. Running `epicontacts::vis_epicontacts()` may overload your session!
+
+::::::::::::::::::
+
+:::::::::::::::::: solution
+
+```{r,message=FALSE,warning=FALSE}
+## first, make an epicontacts object
+ebola_contacts <-
+  epicontacts::make_epicontacts(
+    linelist = ebola_sim_clean$linelist,
+    contacts = ebola_sim_clean$contacts
+  )
+
+# count secondary cases
+
+ebola_infector_secondary <- ebola_contacts %>%
+  pluck("contacts") %>%
+  count(from, name = "secondary_cases")
+
+ebola_secondary <- ebola_contacts %>%
+  # extract ids in contact *and* linelist using "which" argument
+  epicontacts::get_id(which = "all") %>%
+  # transform vector to dataframe to use left_join()
+  enframe(name = NULL, value = "from") %>%
+  # join count secondary cases per infectee
+  left_join(ebola_infector_secondary) %>%
+  # infectee with missing secondary cases are replaced with zero
+  replace_na(
+    replace = list(secondary_cases = 0)
+  )
+
+## plot the distribution
+ebola_secondary %>%
+  ggplot(aes(secondary_cases)) +
+  geom_histogram(binwidth = 1) +
+  labs(
+    x = "Number of secondary cases",
+    y = "Frequency"
+  )
```

Comment:
May I suggest beautifying this plot slightly with custom fill and line attributes?

Reply:
fair suggestion that may apply to other plots here. I'm trying to keep plots with minimal format lines because learners will need to reuse them to visualize the package outputs. Long ggplot lines may cause a blocker into the lesson aims. What we must apply is homogeneity across plots in all tutorials, so this is a pending issue. filled in #63